From bc58ea7c9bc7d1d2654b52a8ada80765698439ab Mon Sep 17 00:00:00 2001
From: Thomas Noack <thomas.noack@iris-sensing.com>
Date: Tue, 6 May 2025 09:48:36 +0200
Subject: [PATCH 4/5] sdbus-c++: upgrade to 2.0.0 release

backported commits:
- 6a8eb25f: sdbus-c++: upgrade to 2.0.0 release
- eb30a6b9: sdbus-c++: Fix build and upgrade to latest git
- aa1a4a8c: sdbus-c++,sdbus-c++-tools: Upgrade to 1.5.0 release
- b1d55efe: sdbus-c++: Install ptests into PTEST_PATH
- cdac7e90: sdbus-c++-tools: Upgrade to 1.4.0
- a098dd0a: sdbus-c++: upgrade 1.3.0 -> 1.4.0
- d88ac90b: feat: bump sdbus-c++ up to v1.3.0
- d3280490: sdbus-c++: Disable ccache
- 842e59e9: sdbus-c++-tools: Upgrade sdbus-c++-tools 1.1.0 to 1.2.0
- d5d57ee1: sdbus-c++: Upgrade sdbus-c++ 1.1.0 to 1.2.0
- 973c0c51: sdbus-c++: Bump version from 1.00 to 1.1.0

Upstream-Status: Backport [meta-openembedded:master, consolidated]
---
 .../sdbus-c++/sdbus-c++-1.0.0/run-ptest       | 15 -----------
 ...ools_1.1.0.bb => sdbus-c++-tools_2.0.0.bb} |  2 +-
 .../sdbus-c++/sdbus-c++/run-ptest             |  6 +++++
 ...{sdbus-c++_1.0.0.bb => sdbus-c++_2.0.0.bb} | 26 ++++++++++---------
 4 files changed, 21 insertions(+), 28 deletions(-)
 delete mode 100755 meta-oe/recipes-core/sdbus-c++/sdbus-c++-1.0.0/run-ptest
 rename meta-oe/recipes-core/sdbus-c++/{sdbus-c++-tools_1.1.0.bb => sdbus-c++-tools_2.0.0.bb} (88%)
 create mode 100755 meta-oe/recipes-core/sdbus-c++/sdbus-c++/run-ptest
 rename meta-oe/recipes-core/sdbus-c++/{sdbus-c++_1.0.0.bb => sdbus-c++_2.0.0.bb} (68%)

diff --git a/meta-oe/recipes-core/sdbus-c++/sdbus-c++-1.0.0/run-ptest b/meta-oe/recipes-core/sdbus-c++/sdbus-c++-1.0.0/run-ptest
deleted file mode 100755
index f6ade0c7e4..0000000000
--- a/meta-oe/recipes-core/sdbus-c++/sdbus-c++-1.0.0/run-ptest
+++ /dev/null
@@ -1,15 +0,0 @@
-#!/bin/sh
-set -e
-set -o pipefail
-
-SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
-
-${SCRIPTPATH}/../tests/sdbus-c++-unit-tests 2>&1 | \
-sed -r 's/^\[\s+OK\s+\] (.*) \([0-9]+\sms\)$/OK: \1 /' | \
-sed -r 's/^\[\s+FAILED\s+\] (.*) \([0-9]+\sms\)$/FAILED: \1 /' | \
-awk '{if ($1 == "OK:" || $1 == "FAILED:") {print $0}}'
-
-${SCRIPTPATH}/../tests/sdbus-c++-integration-tests 2>&1 | \
-sed -r 's/^\[\s+OK\s+\] (.*) \([0-9]+\sms\)$/OK: \1 /' | \
-sed -r 's/^\[\s+FAILED\s+\] (.*) \([0-9]+\sms\)$/FAILED: \1 /' | \
-awk '{if ($1 == "OK:" || $1 == "FAILED:") {print $0}}' 
diff --git a/meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_1.1.0.bb b/meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_2.0.0.bb
similarity index 88%
rename from meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_1.1.0.bb
rename to meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_2.0.0.bb
index 01f66c3f23..af112ced88 100644
--- a/meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_1.1.0.bb
+++ b/meta-oe/recipes-core/sdbus-c++/sdbus-c++-tools_2.0.0.bb
@@ -8,7 +8,7 @@ inherit cmake
 
 DEPENDS += "expat"
 
-SRCREV = "bca8e81037766a0454740c40307eea32831c101f"
+SRCREV = "e62472b210d9e7f06a5e611c23471d414c99a99c"
 SRC_URI = "git://github.com/Kistler-Group/sdbus-cpp.git;protocol=https;branch=master;subpath=tools"
 
 S = "${WORKDIR}/tools"
diff --git a/meta-oe/recipes-core/sdbus-c++/sdbus-c++/run-ptest b/meta-oe/recipes-core/sdbus-c++/sdbus-c++/run-ptest
new file mode 100755
index 0000000000..5e285b39e0
--- /dev/null
+++ b/meta-oe/recipes-core/sdbus-c++/sdbus-c++/run-ptest
@@ -0,0 +1,6 @@
+#!/bin/sh
+set -e
+
+./sdbus-c++-unit-tests 2>&1 && echo "PASS: sdbus-c++-unit-tests" || echo "FAIL: sdbus-c++-unit-tests"
+
+./sdbus-c++-integration-tests 2>&1 && echo "PASS: sdbus-c++-integration-tests" || echo "FAIL: sdbus-c++-integration-tests"
diff --git a/meta-oe/recipes-core/sdbus-c++/sdbus-c++_1.0.0.bb b/meta-oe/recipes-core/sdbus-c++/sdbus-c++_2.0.0.bb
similarity index 68%
rename from meta-oe/recipes-core/sdbus-c++/sdbus-c++_1.0.0.bb
rename to meta-oe/recipes-core/sdbus-c++/sdbus-c++_2.0.0.bb
index 6fd826cbbd..2b6bb1011a 100644
--- a/meta-oe/recipes-core/sdbus-c++/sdbus-c++_1.0.0.bb
+++ b/meta-oe/recipes-core/sdbus-c++/sdbus-c++_2.0.0.bb
@@ -10,20 +10,20 @@ inherit cmake pkgconfig systemd ptest
 
 PACKAGECONFIG ??= "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'with-external-libsystemd', 'with-builtin-libsystemd', d)} \
                    ${@bb.utils.contains('PTEST_ENABLED', '1', 'with-tests', '', d)}"
-PACKAGECONFIG[with-builtin-libsystemd] = ",,sdbus-c++-libsystemd,libcap"
+PACKAGECONFIG[with-builtin-libsystemd] = ",,sdbus-c++-libsystemd,libcap,basu"
 PACKAGECONFIG[with-external-libsystemd] = ",,systemd,libsystemd"
-PACKAGECONFIG[with-tests] = "-DBUILD_TESTS=ON -DTESTS_INSTALL_PATH=${libdir}/${BPN}/tests,-DBUILD_TESTS=OFF,googletest gmock"
+PACKAGECONFIG[with-tests] = "-DSDBUSCPP_BUILD_TESTS=ON -DSDBUSCPP_INSTALL_TESTS=ON -DSDBUSCPP_TESTS_INSTALL_PATH=${PTEST_PATH},-DSDBUSCPP_BUILD_TESTS=OFF,googletest gmock"
 
 DEPENDS += "expat"
 
-SRCREV = "65782bbf435d91e5eb6af9ddab110df406a824a9"
+PV .= "+git"
+SRCREV = "e62472b210d9e7f06a5e611c23471d414c99a99c"
+SRC_URI = "git://github.com/Kistler-Group/sdbus-cpp.git;protocol=https;branch=master \
+           file://run-ptest"
 
-SRC_URI = "git://github.com/Kistler-Group/sdbus-cpp.git;protocol=https;branch=master"
-SRC_URI += "file://run-ptest"
-
-EXTRA_OECMAKE = "-DBUILD_CODE_GEN=OFF \
-                 -DBUILD_DOC=ON \
-                 -DBUILD_DOXYGEN_DOC=OFF"
+EXTRA_OECMAKE = "-DSDBUSCPP_BUILD_CODEGEN=OFF \
+                 -DSDBUSCPP_BUILD_DOCS=ON \
+                 -DSDBUSCPP_BUILD_DOXYGEN_DOCS=OFF"
 
 S = "${WORKDIR}/git"
 
@@ -39,11 +39,13 @@ do_install:append() {
     fi
 }
 
-PTEST_PATH = "${libdir}/${BPN}/ptest"
 do_install_ptest() {
-    install -d ${D}${PTEST_PATH}
-    cp -r ${B}/tests/sdbus-c++-unit-tests  ${D}${PTEST_PATH}
+    DESTDIR='${D}' cmake_runcmake_build --target tests/install
 }
 
 FILES:${PN}-ptest =+ "${sysconfdir}/dbus-1/system.d/"
 FILES:${PN}-dev += "${bindir}/sdbus-c++-xml2cpp"
+
+RDEPENDS:${PN}-ptest += "dbus"
+# It adds -isystem which is spurious, no idea where it gets it from
+CCACHE_DISABLE = "1"
